heat/core/manipulations.py

```diff
@@ -1036,10 +1036,14 @@ def unique(a, sorted=False, return_inverse=False, axis=None):
            [3, 1]])
     """
     if a.split is None:
-        # Trivial case, result can just be forwarded
-        return torch.unique(
+        torch_output = torch.unique(
             a._DNDarray__array, sorted=sorted, return_inverse=return_inverse, dim=axis
         )
+        if isinstance(torch_output, tuple):
+            heat_output = tuple(factories.array(i, dtype=a.dtype) for i in torch_output)
```

Comment:
different number of arguments?

Reply:
also the array calls should have a device parameter

Reply:
and split